### PR TITLE
Reset `url` before updating page store

### PR DESCRIPTION
**Pull request by Rich-Harris**
Alternative to #4773. We can keep the existing caching logic, we just need to make sure `url` isn't stale

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

File: packages/kit/test/apps/basics/test/test.js [hunk 1]
```diff
@@ -1989,6 +1989,108 @@ test.describe.parallel('Redirects', () => {
...
+			await page.goto('/routing/a');
+
+			await app.prefetch('/routing/prefetched/hash-route#please-dont-show-me');
+			await app.prefetch('/routing/prefetched/hash-route#please-dont-show-me-jr');
+			await app.goto('/routing/prefetched/hash-route#please-dont-show-me');
+			await expect(page.locator('p')).toHaveText('Loaded 1 times.');
+		}
+	});
+
+	test('successfully preloads same route with different hashes many times', async ({
```
Comment (tcc-sejohnson) [inline packages/kit/test/apps/basics/test/test.js:2073]: Should probably remove this test since there's no promise-wrapping danger now.

**Comment by Rich-Harris** [comment]
Ah whoops, I deleted the 'successfully preloads same route with different hashes many times' test in response to your since-deleted comment — was that a mistake?

**Comment by tcc-sejohnson** [comment]
@Rich-Harris

Nope, all good -- I referenced the wrong lines in the comment and couldn't figure out how to update the reference, so I was just going to re-comment with the correct line references... you were just a bit too fast on the uptake. :)

**Comment by tcc-sejohnson** [comment]
@Rich-Harris Quick note: Remember to close #4768 after this is merged -- the issue is no longer linked to the PR.

**Pull request merged**